src/mito2/src/compaction.rs

```diff
@@ -584,7 +583,7 @@ impl<'a> CompactionSstReaderBuilder<'a> {
         let mut scan_input = ScanInput::new(self.sst_layer, ProjectionMapper::all(&self.metadata)?)
             .with_files(self.inputs.to_vec())
             .with_append_mode(self.append_mode)
-            .with_cache(self.cache)
```

Comment:
If we don't pass the cache manager to the compaction reader, we can't use the  write cache to speed up the compaction and reduce the API calls to the object store. @v0y4g3r